acmw-db-app/components/FormComponents/CompanySearchInput.jsx

```diff
@@ -1,12 +1,54 @@
 import SelectInput from './SelectInput'
+import {useState} from 'react'
+import styles from '../../styles/components/FormComponents.module.css'
+import MultiSelectInput from './MultiSelectInput'
 
-const CompanySearchInput = () => {
+// I'm so bad at understanding promises/fetch() so please point out if I'm doing something wrong
 
-    // Eventually this will connect to the database of companies and provide a full list
-    const sampleCompanies = ["Nimbis Services", "Target", "Capital One", "Facebook"]
+const getCompanyInfo = async (input) => {
+    const url = '/api/company/byString?input=' + input
+    const response = await fetch(url, {
+        method: 'GET', 
+        mode: 'cors', 
+        cache: 'no-cache', 
+        credentials: 'same-origin', 
+        headers: {
+        'Content-Type': 'application/json'
+        },
+        redirect: 'follow', 
+        referrerPolicy: 'no-referrer', 
+    })
+    return response.json() 
+}
+
+const CompanySearchInput = (props) => {
+
+    const [companyList, setCompanyList] = useState([])
+    const [isText, setIsText] = useState(false) // Has the user entered text
+
+    const filterFunction = (event) => {
+        if (event.target.value != "") {
+            setIsText(true)
+            getCompanyInfo(event.target.value).then((data) => {
+                const tempList = []
+                for (var i in data) {
+                    tempList.push(data[i]["cname"])
+                }
+                setCompanyList(tempList)
+            })
+        } else {
+            setIsText(false)
+            setCompanyList([])
+        }
+    }
 
     return (
-        <SelectInput options={sampleCompanies} label={"Select Company"}/>
+        <div>
+            <label className={styles.label}>Company Search</label>
+            <input autoComplete="off" type="text" className={styles.field} placeholder="Search..." id="companySearch" onKeyDown={filterFunction}></input>
```

Comment:
`onKeyDown` and `onKeyPress` fire before `onChange` so the text it searches is 1 character behind (and does not take into account deletes for some reason idk).

but changing `onKeyDown={filterFunction}` to `onChange={filterFunction}` should fix it